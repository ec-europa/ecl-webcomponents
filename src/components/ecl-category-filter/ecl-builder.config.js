const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-category-filter/category-filter.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-category-filter-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-category-filter-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-category-filter-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-category-filter-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-category-filter-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
