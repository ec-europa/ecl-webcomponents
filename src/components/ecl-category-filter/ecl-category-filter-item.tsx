import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-category-filter-item',
  shadow: false,
})

export class EclCategoryFilterItem {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() label: string;
  @Prop() path: string;
  @Prop() level: number;
  @Prop() subItems: boolean = false;
  @Prop() expanded: boolean = false;
  @Prop() lastClicked: HTMLElement;

  clickedItem(ev): void {
    this.lastClicked = ev.target.closest('.ecl-category-filter__list-item');
  }

  getClass(): string {
    return [
      `ecl-category-filter__list-item`,
      `sc-ecl-category-filter-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  getLinkClass(): string {
    const linkClasses = [
      `ecl-category-filter__item`,
      `sc-ecl-category-filter-${this.theme}`,
      `ecl-category-filter__item--level-${this.level}`,
      this.styleClass
    ];

    if (this.subItems) {
      linkClasses.push(`ecl-category-filter__item--has-children`);
    }

    return linkClasses.join(' ');
  }

  render() {
    const elAttrs = {};
    if (this.subItems) {
      elAttrs['aria-expanded'] =  "false";
    }
    if (this.expanded) {
      elAttrs['aria-expanded'] =  "true";
    }
    return (
      <li
        class={this.getClass()}
        {...elAttrs}
        onClick={ev => this.clickedItem(ev)}
      >
      { this.path ?
        <ecl-link
          style-class={this.getLinkClass()}
          path={this.path}
        >
          {this.label}
          {this.subItems ?
            <ecl-icon
              style-class={`ecl-category-filter__item-icon sc-ecl-category-filter-${this.theme}`}
              icon={this.level === 1 ? 'corner-arrow' : 'solid-arrow'}
              size={this.level === 1 ? 'xs' : 'm'}
              transform={this.level === 1 ? 'rotate-180' : 'rotate-90'}
              slot={this.level === 1 ? 'icon-after' : 'icon-before'}
            ></ecl-icon> : '' 
          }
        </ecl-link> : '' 
      }
      { !this.path ? this.label : '' }
      { this.subItems ?
        <ul class={`ecl-category-filter__list sc-ecl-category-filter-${this.theme}`}>
          <slot></slot>
        </ul> : ''
      }
      </li>
    )
  }
}
