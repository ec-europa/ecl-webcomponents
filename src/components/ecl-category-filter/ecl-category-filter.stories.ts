export default {
  title: 'Components/category-filter',
};

const Template = (args) => 
`<ecl-category-filter
  theme="${args.theme}"
  ecl-script
>
  <ecl-category-filter-item
    path="/example.html"
    label="Item 1 label"
    level="1"
  >
  </ecl-category-filter-item>
  <ecl-category-filter-item
    path="/example.html"
    label="Item 2 label"
    level="1"
    sub-items
    current
  >
    <ecl-category-filter-item
      path="/example.html"
      label="Item 2.1"
      level="2"
      sub-items
      current
    >
      <ecl-category-filter-item
        label="Item 2.1.1"
        path="/example.html"
        level="3"
      ></ecl-category-filter-item>
      <ecl-category-filter-item
        label="Item 2.1.2 "
        path="/example.html"
        level="3"
        sub-items
      >
        <ecl-category-filter-item
          label="Item 2.1.2 a"
          path="/example.html"
          level="4"
        >
        </ecl-category-filter-item>
        <ecl-category-filter-item
          label="Item 2.1.2 b"
          path="/example.html"
          level="4"
        >
        </ecl-category-filter-item>
        <ecl-category-filter-item
          label="Item 2.1.2 c"
          path="/example.html"
          level="4"
        >
        </ecl-category-filter-item>
      </ecl-category-filter-item>
      <ecl-category-filter-item
        path="/example.html"
        label="Item 2.1.3"
        level="3"
      >
      </ecl-category-filter-item>
      <ecl-category-filter-item
        path="/example.html"
        label="Item 2.1.4"
        level="3"
      >
      </ecl-category-filter-item>
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 2.2"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 2.3"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 2.4"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 2.5"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 2.6"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 2.7"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
  </ecl-category-filter-item>
  <ecl-category-filter-item
    path="/example.html"
    label="Item 3 with a very long label"
    level="1"
    sub-items
  >
    <ecl-category-filter-item
      label="Item 3.1"
      path="/example.html"
      level="2"
      sub-items
    >
      <ecl-category-filter-item
        label="Item 3.1.1"
        path="/example.html"
        level="3"
      >
      </ecl-category-filter-item>
      <ecl-category-filter-item
        label="Item 3.1.2"
        path="/example.html"
        level="3"
        sub-items
      >
        <ecl-category-filter-item
          label="Item 3.1.2 a"
          path="/example.html"
          level="4"
        >
        </ecl-category-filter-item>
        <ecl-category-filter-item
          label="Item 3.1.2 b"
          path="/example.html"
          level="4"
        >
        </ecl-category-filter-item>
        <ecl-category-filter-item
          label="Item 3.1.2 c"
          path="/example.html"
          level="4"
        >
        </ecl-category-filter-item>
      </ecl-category-filter-item>
      <ecl-category-filter-item
        label="Item 3.1.3"
        path="/example.html"
        level="3"
      >
      </ecl-category-filter-item>
      <ecl-category-filter-item
        label="Item 3.1.4"
        path="/example.html"
        level="3"
      >
      </ecl-category-filter-item>
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 3.2"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
    <ecl-category-filter-item
      label="Item 3.3"
      path="/example.html"
      level="2"
    >
    </ecl-category-filter-item>
  </ecl-category-filter-item>
  <ecl-category-filter-item
    label="Item 4"
    path="/example.html"
    level="1"
  >
  </ecl-category-filter-item>
</ecl-category-filter>`;

export const CategoryFilter = Template.bind({});
CategoryFilter.storyName = 'default';
