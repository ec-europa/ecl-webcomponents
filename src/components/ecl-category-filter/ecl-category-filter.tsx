import { Component, Prop, h, Element, getAssetPath } from '@stencil/core';
declare const ECL: any;

@Component({
  tag: 'ecl-category-filter',
  styleUrls: {
    ec: './build/styles/ecl-category-filter-ec.css',
    eu: './build/styles/ecl-category-filter-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclCategoryFilter {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() eclScript: boolean = false;

  getClass(): string {
    return [
      `ecl-category-filter`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    const items = Array.from(this.el.querySelectorAll('.ecl-category-filter__item--level-1'));
    if (items[0]) {
      const lastItem = items[items.length - 1] as HTMLElement;
      lastItem.parentElement.parentElement.style.border = 'none';
    }
    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-category-filter-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const categoryFilter = new ECL.CategoryFilter(this.el.firstElementChild);
        categoryFilter.init();
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <nav
        class={this.getClass()}
        data-ecl-category-filter
      >
        <ul class="ecl-category-filter__list">
          <slot></slot>
        </ul>
      </nav>
    )
  }
}
