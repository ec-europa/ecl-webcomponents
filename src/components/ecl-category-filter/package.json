{
  "name": "@ecl/web-components-category-filter",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.0.0",
  "description": "ECL Category Filter",
  "main": "ecl-category-filter.tsx",
  "private": true,
  "scripts": {
    "build:scripts": "cross-env ecl-builder scripts",
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.0",
    "@ecl/vanilla-component-category-filter": "3.7.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}