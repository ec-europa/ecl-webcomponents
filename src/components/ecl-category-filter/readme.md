# ecl-category-filter-subitem



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description | Type          | Default     |
| ------------- | ------------- | ----------- | ------------- | ----------- |
| `expanded`    | `expanded`    |             | `boolean`     | `false`     |
| `label`       | `label`       |             | `string`      | `undefined` |
| `lastClicked` | --            |             | `HTMLElement` | `undefined` |
| `level`       | `level`       |             | `number`      | `undefined` |
| `path`        | `path`        |             | `string`      | `undefined` |
| `styleClass`  | `style-class` |             | `string`      | `''`        |
| `subItems`    | `sub-items`   |             | `boolean`     | `false`     |
| `theme`       | `theme`       |             | `string`      | `'ec'`      |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-category-filter-item --> ecl-link
  ecl-category-filter-item --> ecl-icon
  ecl-link --> ecl-icon
  style ecl-category-filter-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
