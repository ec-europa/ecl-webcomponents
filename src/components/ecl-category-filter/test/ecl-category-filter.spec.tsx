import { newSpecPage } from '@stencil/core/testing';
import { EclCategoryFilter } from '../ecl-category-filter';
import { EclCategoryFilterItem } from '../ecl-category-filter-item';

describe('ecl-category-filter', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclCategoryFilter, EclCategoryFilterItem],
      html: `<ecl-category-filter
        theme="ec"
        ecl-script
      >
        <ecl-category-filter-item
          path="/example"
          label="Item 1 label"
          level="1"
        >
        </ecl-category-filter-item>
        <ecl-category-filter-item
          path="/example"
          label="Item 2 label"
          level="1"
          sub-items
        >
          <ecl-category-filter-item
            path="/example"
            label="Item 2.1"
            level="2"
            sub-items
          >
            <ecl-category-filter-item
              label="Item 2.1.1"
              path="/example"
              level="3"
            ></ecl-category-filter-item>
            <ecl-category-filter-item
              label="Item 2.1.2 "
              path="/example"
              level="3"
              sub-items
            >
              <ecl-category-filter-item
                label="Item 2.1.2 a"
                path="/example"
                level="4"
              >
              </ecl-category-filter-item>
              <ecl-category-filter-item
                label="Item 2.1.2 b"
                path="/example"
                level="4"
              >
              </ecl-category-filter-item>
              <ecl-category-filter-item
                label="Item 2.1.2 c"
                path="/example"
                level="4"
              >
              </ecl-category-filter-item>
            </ecl-category-filter-item>
            <ecl-category-filter-item
              path="/example"
              label="Item 2.1.3"
              level="3"
            >
            </ecl-category-filter-item>
            <ecl-category-filter-item
              path="/example"
              label="Item 2.1.4"
              level="3"
            >
            </ecl-category-filter-item>
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 2.2"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 2.3"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 2.4"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 2.5"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 2.6"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 2.7"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
        </ecl-category-filter-item>
        <ecl-category-filter-item
          path="/example"
          label="Item 3 with a very long label"
          level="1"
          sub-items
        >
          <ecl-category-filter-item
            label="Item 3.1"
            path="/example"
            level="2"
            sub-items
          >
            <ecl-category-filter-item
              label="Item 3.1.1"
              path="/example"
              level="3"
            >
            </ecl-category-filter-item>
            <ecl-category-filter-item
              label="Item 3.1.2"
              path="/example"
              level="3"
              sub-items
            >
              <ecl-category-filter-item
                label="Item 3.1.2 a"
                path="/example"
                level="4"
              >
              </ecl-category-filter-item>
              <ecl-category-filter-item
                label="Item 3.1.2 b"
                path="/example"
                level="4"
              >
              </ecl-category-filter-item>
              <ecl-category-filter-item
                label="Item 3.1.2 c"
                path="/example"
                level="4"
              >
              </ecl-category-filter-item>
            </ecl-category-filter-item>
            <ecl-category-filter-item
              label="Item 3.1.3"
              path="/example"
              level="3"
            >
            </ecl-category-filter-item>
            <ecl-category-filter-item
              label="Item 3.1.4"
              path="/example"
              level="3"
            >
            </ecl-category-filter-item>
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 3.2"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
          <ecl-category-filter-item
            label="Item 3.3"
            path="/example"
            level="2"
          >
          </ecl-category-filter-item>
        </ecl-category-filter-item>
        <ecl-category-filter-item
          label="Item 4"
          path="/example"
          level="1"
        >
        </ecl-category-filter-item>
      </ecl-category-filter>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
