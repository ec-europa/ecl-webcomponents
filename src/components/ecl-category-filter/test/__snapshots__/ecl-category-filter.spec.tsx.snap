// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-category-filter renders 1`] = `
<ecl-category-filter ecl-script="" theme="ec">
  <!---->
  <nav class="ecl-category-filter" data-ecl-category-filter>
    <ul class="ecl-category-filter__list">
      <ecl-category-filter-item label="Item 1 label" level="1" path="/example">
        <!---->
        <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
          <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-1 ">
            Item 1 label
          </ecl-link>
        </li>
      </ecl-category-filter-item>
      <ecl-category-filter-item label="Item 2 label" level="1" path="/example" sub-items="">
        <!---->
        <li aria-expanded="false" class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
          <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-1  ecl-category-filter__item--has-children">
            Item 2 label
            <ecl-icon icon="corner-arrow" size="xs" slot="icon-after" style-class="ecl-category-filter__item-icon sc-ecl-category-filter-ec" transform="rotate-180"></ecl-icon>
          </ecl-link>
          <ul class="ecl-category-filter__list sc-ecl-category-filter-ec">
            <ecl-category-filter-item label="Item 2.1" level="2" path="/example" sub-items="">
              <!---->
              <li aria-expanded="false" class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2  ecl-category-filter__item--has-children">
                  Item 2.1
                  <ecl-icon icon="solid-arrow" size="m" slot="icon-before" style-class="ecl-category-filter__item-icon sc-ecl-category-filter-ec" transform="rotate-90"></ecl-icon>
                </ecl-link>
                <ul class="ecl-category-filter__list sc-ecl-category-filter-ec">
                  <ecl-category-filter-item label="Item 2.1.1" level="3" path="/example">
                    <!---->
                    <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3 ">
                        Item 2.1.1
                      </ecl-link>
                    </li>
                  </ecl-category-filter-item>
                  <ecl-category-filter-item label="Item 2.1.2 " level="3" path="/example" sub-items="">
                    <!---->
                    <li aria-expanded="false" class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3  ecl-category-filter__item--has-children">
                        Item 2.1.2
                        <ecl-icon icon="solid-arrow" size="m" slot="icon-before" style-class="ecl-category-filter__item-icon sc-ecl-category-filter-ec" transform="rotate-90"></ecl-icon>
                      </ecl-link>
                      <ul class="ecl-category-filter__list sc-ecl-category-filter-ec">
                        <ecl-category-filter-item label="Item 2.1.2 a" level="4" path="/example">
                          <!---->
                          <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                            <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-4 ">
                              Item 2.1.2 a
                            </ecl-link>
                          </li>
                        </ecl-category-filter-item>
                        <ecl-category-filter-item label="Item 2.1.2 b" level="4" path="/example">
                          <!---->
                          <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                            <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-4 ">
                              Item 2.1.2 b
                            </ecl-link>
                          </li>
                        </ecl-category-filter-item>
                        <ecl-category-filter-item label="Item 2.1.2 c" level="4" path="/example">
                          <!---->
                          <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                            <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-4 ">
                              Item 2.1.2 c
                            </ecl-link>
                          </li>
                        </ecl-category-filter-item>
                      </ul>
                    </li>
                  </ecl-category-filter-item>
                  <ecl-category-filter-item label="Item 2.1.3" level="3" path="/example">
                    <!---->
                    <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3 ">
                        Item 2.1.3
                      </ecl-link>
                    </li>
                  </ecl-category-filter-item>
                  <ecl-category-filter-item label="Item 2.1.4" level="3" path="/example">
                    <!---->
                    <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3 ">
                        Item 2.1.4
                      </ecl-link>
                    </li>
                  </ecl-category-filter-item>
                </ul>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 2.2" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 2.2
                </ecl-link>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 2.3" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 2.3
                </ecl-link>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 2.4" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 2.4
                </ecl-link>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 2.5" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 2.5
                </ecl-link>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 2.6" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 2.6
                </ecl-link>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 2.7" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 2.7
                </ecl-link>
              </li>
            </ecl-category-filter-item>
          </ul>
        </li>
      </ecl-category-filter-item>
      <ecl-category-filter-item label="Item 3 with a very long label" level="1" path="/example" sub-items="">
        <!---->
        <li aria-expanded="false" class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
          <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-1  ecl-category-filter__item--has-children">
            Item 3 with a very long label
            <ecl-icon icon="corner-arrow" size="xs" slot="icon-after" style-class="ecl-category-filter__item-icon sc-ecl-category-filter-ec" transform="rotate-180"></ecl-icon>
          </ecl-link>
          <ul class="ecl-category-filter__list sc-ecl-category-filter-ec">
            <ecl-category-filter-item label="Item 3.1" level="2" path="/example" sub-items="">
              <!---->
              <li aria-expanded="false" class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2  ecl-category-filter__item--has-children">
                  Item 3.1
                  <ecl-icon icon="solid-arrow" size="m" slot="icon-before" style-class="ecl-category-filter__item-icon sc-ecl-category-filter-ec" transform="rotate-90"></ecl-icon>
                </ecl-link>
                <ul class="ecl-category-filter__list sc-ecl-category-filter-ec">
                  <ecl-category-filter-item label="Item 3.1.1" level="3" path="/example">
                    <!---->
                    <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3 ">
                        Item 3.1.1
                      </ecl-link>
                    </li>
                  </ecl-category-filter-item>
                  <ecl-category-filter-item label="Item 3.1.2" level="3" path="/example" sub-items="">
                    <!---->
                    <li aria-expanded="false" class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3  ecl-category-filter__item--has-children">
                        Item 3.1.2
                        <ecl-icon icon="solid-arrow" size="m" slot="icon-before" style-class="ecl-category-filter__item-icon sc-ecl-category-filter-ec" transform="rotate-90"></ecl-icon>
                      </ecl-link>
                      <ul class="ecl-category-filter__list sc-ecl-category-filter-ec">
                        <ecl-category-filter-item label="Item 3.1.2 a" level="4" path="/example">
                          <!---->
                          <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                            <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-4 ">
                              Item 3.1.2 a
                            </ecl-link>
                          </li>
                        </ecl-category-filter-item>
                        <ecl-category-filter-item label="Item 3.1.2 b" level="4" path="/example">
                          <!---->
                          <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                            <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-4 ">
                              Item 3.1.2 b
                            </ecl-link>
                          </li>
                        </ecl-category-filter-item>
                        <ecl-category-filter-item label="Item 3.1.2 c" level="4" path="/example">
                          <!---->
                          <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                            <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-4 ">
                              Item 3.1.2 c
                            </ecl-link>
                          </li>
                        </ecl-category-filter-item>
                      </ul>
                    </li>
                  </ecl-category-filter-item>
                  <ecl-category-filter-item label="Item 3.1.3" level="3" path="/example">
                    <!---->
                    <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3 ">
                        Item 3.1.3
                      </ecl-link>
                    </li>
                  </ecl-category-filter-item>
                  <ecl-category-filter-item label="Item 3.1.4" level="3" path="/example">
                    <!---->
                    <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                      <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-3 ">
                        Item 3.1.4
                      </ecl-link>
                    </li>
                  </ecl-category-filter-item>
                </ul>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 3.2" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 3.2
                </ecl-link>
              </li>
            </ecl-category-filter-item>
            <ecl-category-filter-item label="Item 3.3" level="2" path="/example">
              <!---->
              <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
                <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-2 ">
                  Item 3.3
                </ecl-link>
              </li>
            </ecl-category-filter-item>
          </ul>
        </li>
      </ecl-category-filter-item>
      <ecl-category-filter-item label="Item 4" level="1" path="/example">
        <!---->
        <li class="ecl-category-filter__list-item sc-ecl-category-filter-ec">
          <ecl-link path="/example" style-class="ecl-category-filter__item sc-ecl-category-filter-ec ecl-category-filter__item--level-1 ">
            Item 4
          </ecl-link>
        </li>
      </ecl-category-filter-item>
    </ul>
  </nav>
</ecl-category-filter>
`;
